chapters/kmodes/main.tex

```diff
@@ -1 +1,1049 @@
+
+Both of the initialisation methods described in this section have a greedy
+component. Cao's method essentially chooses the point with highest density that
+has not already been chosen whilst forcing some separation between the set of
+initial modes. In the case of Huang's, the greediness only comes toward the end
+of the method. When the set of potential modes is identified, it is replaced by
+a set of instances in the dataset. Specifically, this means that in any
+practical implementation of this method, the order in which a set of potential
+modes is iterated over can affect the set of initial modes. Thus, there is no
+guarantee of consistency. The same is true for any arbitrary tie breaks that may
+occur.
+
+The initialisation proposed in this chapter extends Huang's method to be
+order-invariant in the final allocation. Doing so eliminates its greedy
+component and provides a more intuitive starting point for the \(k\)-modes
+algorithm by constructing and solving a matching game between the set of
+potential modes and some subset of the data.
+
+Matching games are a construct from game theory that allow for the allocation of
+partnerships between agents (players) such that no pair of players is rationally
+envious of any partnership. Appendix~\ref{app:matching} provides an extensive
+introduction to several fundamental matching games, the most commonly found of
+which is the stable marriage problem (SM). SM, like many matching games,
+involves two distinct sets of players (parties), each of whom have a strict
+ranking of the players in the other set. Resolving these player preferences to
+find a matching in which no two players would rather be matched to one another
+than their current matches is called \emph{solving} the game.~\cite{Gale1962}
+formalised this problem and provided an algorithm that guarantees to find an
+stable, party-optimal matching to any instance of SM.
+
+Despite its simplicity, using SM in this scenario would reduce the proposed
+method to Huang's method. This can be seen as follows. Both parties in the game
+must be of the same size, and so \(k\) data points must be considered along with
+the potential modes. The most straightforward way of selecting these points is
+to allow each potential mode to add its closest point that has not already been
+selected. Then, upon solving the game, and up to an arbitrary breaking of any
+ties in the preference lists, each potential mode is assigned to its chosen data
+point.
+
+The particular constraints of this case --- where the \(k\) potential modes must
+be allocated to a nearby and unique data point --- mirror those of the
+hospital-resident assignment problem (HR) discussed in
+Appendix~\ref{app:matching}. In particular, this scenario can be modelled by an
+instance of HR where each hospital has a capacity of one. A contextual summary
+of the remaining HR game components and those of the proposed initialisation is
+given in Table~\ref{tab:components}.
```

Comment:
This is the k-modes/matching overlap section @drvinceknight

Reply:
Actually, down to line ~480

Reply:
Re-reading this section, I can see it needs fleshing out some (even basic notation and nomenclature would do.)

Otherwise this section relies on the appendix totally, which I'm guessing isn't the idea

Reply:
Great thanks will read through after JJs bath: does it correspond to 0b2e81d? (no problem if not will look through the lines you pointed out)

Reply:
Yes it does @drvinceknight 👍

Reply:
I've addressed these avenues now in 9dc3969 and 1e6978c @drvinceknight

Reply:
Just read through: looks good to me :+1: